src/timeout.py

```diff
@@ -20,96 +20,164 @@
 from base import Service
 
 
-class Timeout(Service):
+class TimeoutService(Service):
 
-    def __init__(self, configs, args):
-        super().__init__(configs, args, 'timeout')
+    def __init__(self, configs, args, name):
+        super().__init__(configs, args, name)
         self._poll_period = args.poll_period
 
-    def _set_node_result(self, node):
-        """Set node result of timed out nodes
-        If the node is in 'running' state, set the node result to 'incomplete'.
-        When the timed out node is not in 'running' state, set the node result
-        to 'fail' if one of the child nodes failed. Otherwise, set the result
-        to 'pass'.
-        """
-        if node['state'] == 'running':
-            node['result'] = 'incomplete'
-            return node
-        failed_child_nodes = self._db.count_nodes({
-            'parent': node['_id'],
-            'result': 'fail'
-        })
-        node['result'] = 'fail' if failed_child_nodes else 'pass'
-        return node
-
-    def _update_available_node(self, node, current_time):
-        """Set node state to 'done' if holdoff is expired and all child nodes
-        are completed.
-        Set node state to `closing` if holdoff is expired and at least one of
-        the child nodes is incomplete"""
-        holdoff_time = datetime.fromisoformat(node['holdoff'])
-        if current_time > holdoff_time:
-            self._logger.log_message(logging.INFO,
-                                     f"Holdoff reached for {node['_id']}")
-            total_child_nodes = self._db.count_nodes({
-                'parent': node['_id'],
+    def _get_pending_nodes(self, filters=None):
+        nodes = {}
+        node_filters = filters.copy() if filters else {}
+        for state in ['running', 'available', 'closing']:
```

Comment:
How about declaring an instance variable with `['running', 'available', 'closing']` in `TimeoutService` class as the list is being used in multiple methods?
```
self._pending_node_states = ['running', 'available', 'closing']
```

Reply:
See my replies to @mgalka's comments, I think the definition of the nodes states should be revisited so I didn't want to optimise things here.  But yes we could add this, even as a class variable as it won't change with each instance.

Reply:
OK I've sent an update for that.